android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt

```diff
@@ -803,7 +804,7 @@ class GeckoEngineSession(
      */
     override fun requestAnalysisStatus(
         url: String,
-        onResult: (String) -> Unit,
+        onResult: (ProductAnalysisStatus) -> Unit,
```

Comment:
We should update where [this is integrated in fenix](https://searchfox.org/mozilla-mobile/source/firefox-android/fenix/app/src/main/java/org/mozilla/fenix/shopping/middleware/ReviewQualityCheckService.kt#113). For this PR, we can simply update `continuation.resume(it.status.asEnumOrDefault(AnalysisStatusDto.OTHER))`